### PR TITLE
Markdown

### LINKED ISSUE #16: In the add problem section the markdown editor doesn't show the formula correctly.
https://github.com/Caribbean-Online-Judge/COJ.Web.Frontend/blob/markdown/src/pages/admin/problems/add.page.tsx

### PR DESCRIPTION
**Pull request by Kosmo00**
close #16

### CODE REVIEW

File: src/pages/_app.page.tsx [hunk 1]
```diff
@@ -37,17 +41,20 @@ function MyApp({ Component, pageProps }: AppProps): JSX.Element {
          <Head>
             <title>Caribbean Online Judge</title>
          </Head>
+         <CssBaseline />
          <Provider store={store}>
             <ThemeProvider theme={theme}>
-               {router.pathname.startsWith("/account") ? (
```
Comment (cl8dep) [inline src/pages/_app.page.tsx:42]: Please move this logic to an independent function to get more clear render function

File: src/pages/admin/problems/add.page.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+   TextField,
+} from "./styles"
+import { useServerManager } from "../../../api/server"
+import { ProblemDTO } from "../../../types/ProblemDTO"
+import { Button, CardContent, CardHeader, Divider } from "@mui/material"
+
+const MDEditor = dynamic(() => import("@uiw/react-md-editor"), { ssr: false })
+
+export default function AddProblem(): JSX.Element {
+   const mdKaTeX = `This is to display the
```
Comment (cl8dep) [inline src/pages/admin/problems/add.page.tsx]: Remove this variable

File: src/pages/admin/problems/add.page.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      },
+      text: mdKaTeX,
+      title: "",
+      description: "",
+   })
+
+   const handleUpload = () => {
+      const data = new FormData()
+
+      Object.entries(problemData).forEach((entie) => {
```
Comment (cl8dep) [inline src/pages/admin/problems/add.page.tsx]: Entie? It's should be entry

File: src/pages/admin/problems/add.page.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+         data.append(entie[0], entie[1])
+      })
+
+      serverManager.createProblem(data)
+   }
+
+   const disableButton =
```
Comment (cl8dep) [inline src/pages/admin/problems/add.page.tsx]: The forma should be validate using some solution to do that. For example formstate

**Review by cl8dep — CHANGES_REQUESTED** [review]
We need seriously thinking in a solution to do business logic in another way for example via Redux or Context but the UI components should be independent of its business logic.

File: package.json [hunk 5]
```diff
@@ -22,16 +22,25 @@
...
+    "next-remove-imports": "^1.0.6",
     "react": "17.0.2",
     "react-dom": "17.0.2",
+    "react-hook-form": "^7.22.5",
     "react-lottie-player": "^1.4.1",
+    "react-markdown": "^7.1.1",
     "react-redux": "^7.2.6",
-    "redux-thunk": "^2.4.1"
+    "redux-thunk": "^2.4.1",
+    "remark-gfm": "^3.0.1"
```
Comment (cl8dep) [inline package.json]: What the hell is this?

File: src/pages/admin/problems/add.page.tsx [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+      //    data.append(entry[0], entry[1])
+      // })
+      // serverManager.createProblem(data)
+      const data = getValues()
+      console.log(data)
+   }
+
+   return (
+      <Box>
+         <Controller
```
Comment (cl8dep) [inline src/pages/admin/problems/add.page.tsx]: This logic can be implemented using a self created component to avoid repeat code

**Review by cl8dep — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (jesusr00) [reply package.json]: remark-gfm? 
Not idea, I'll remove it

**Pull request merged**